### DISCUSSION TITLE
MIDI clock from VST host?

### DISCUSSION THREAD

**Discussion by seanjean3000** [plugdata-team/plugdata#241] [Q&A]
I have a patch I'd like to use as a VST instrument that has an internal sequencer, which I would like to sync to the host. Is there a way to do this, maybe with MIDI clock? Or, what might Plug Data receive from the host that could be used this way? (I've tried the MIDI realtime object, and the MIDI clock object, neither seems to receive anything).

**Comment by broadskins** [comment]
Oops, maybe I just asked nearly the same thing in a completely different way! Oh well, hopefully one of us will get an answer!

**Comment by seanjean3000** [comment]
I just found it the other day: [midi.clock]

There's also the [playhead] object to get playstates, bpm, etc from the DAW / host

**Reply by broadskins** [reply]
Okay, I was experimenting with the [midi.clock] but couldn't work out exactly what it is sending - what does it send?

**Reply by seanjean3000** [reply]
You should be seeing 96 bangs per ***quarter note*** coming out of [midi.clock], though I don't remember if I've checked this. 

I read the question you posted; as far as I understand, MIDI clock by convention is 96 pulses per quarter note, not 960, though that is used in software these days. But maybe Pure Data respects the 96ppq. I'm just guessing, I don't know / haven't looked into it.

At 120 BPM though, that still means 5 milliseconds intervals, which I seems pretty reasonable for capturing 'raw' timing?

Edit: I had said 96 bangs per second, but I meant to say per quarter note, sorry!

**Reply by broadskins** [reply]
Yes, seems like a good place to start, I think I've been spoilt with 960ppq!
